regression/apps/mongoose/Makefile

```diff
@@ -4,44 +4,25 @@ MAIN_MAKEFILE := $(firstword $(MAKEFILE_LIST))
 INCLUDE_MAKEFILE := $(lastword $(MAKEFILE_LIST))
 CUR_DIR := $(shell dirname $(realpath $(MAIN_MAKEFILE)))
 BUILD_DIR := $(CUR_DIR)/../../build/initramfs/regression/network
+MONGOOSE_DIR := $(CUR_DIR)/../../build/mongoose
+MONGOOSE_FILES := $(MONGOOSE_DIR)/mongoose.h $(MONGOOSE_DIR)/mongoose.c
+SRC_SERVER := http_server.c $(MONGOOSE_DIR)/mongoose.c
+SRC_CLIENT := http_client.c $(MONGOOSE_DIR)/mongoose.c
 TARGET_SERVER := $(BUILD_DIR)/http_server
 TARGET_CLIENT := $(BUILD_DIR)/http_client
-CC := cc
-CFLAGS := -W -Wall -Wextra -g -I. -DMG_HTTP_DIRLIST_TIME_FMT="%Y/%m/%d %H:%M:%S" -DMG_ENABLE_LINES=1 -DMG_ENABLE_IPV6=1 -DMG_ENABLE_SSI=1
-SRC_SERVER := http_server.c mongoose.c
-SRC_CLIENT := http_client.c mongoose.c
-DEP := mongoose.h
-
-.PHONY: all clean mongoose server client
-
-all: server client
-
-server: $(TARGET_SERVER)
 
-client: $(TARGET_CLIENT)
-
-# Rule to build the http server
-$(TARGET_SERVER): $(SRC_SERVER) | $(BUILD_DIR)
-	$(CC) $(SRC_SERVER) $(CFLAGS) -o $@
+CC := cc
+CFLAGS := -W -Wall -Wextra -g -I. -I$(MONGOOSE_DIR) -DMG_ENABLE_LINES=1
 
-# Rule to build the http client
-$(TARGET_CLIENT): $(SRC_CLIENT) | $(BUILD_DIR)
-	$(CC) $(SRC_CLIENT) $(CFLAGS) -o $@
+.PHONY: all 
+all: $(BUILD_DIR) | $(MONGOOSE_FILES)
```

Comment:
As a rule, please make directories (e.g., `$(BUILD_DIR)`) order-only prerequisites (prerequisites after the `|`) and make files (e.g., `$(MONGOOSE_FILES)`) normal prerequisites (prerequisites before the `|`).

This will prevent accidental rebuilding when the directories "change".

Reply:
Thank you for your suggestions. I will list them after the | to indicate that they should be order-only prerequisites.